src/utils.rs

```diff
@@ -0,0 +1,33 @@
+// A poly-fill for `lazy_cell`
+// Replace with std::sync::LazyLock when https://github.com/rust-lang/rust/issues/109736 is stablized.
+
+use std::cell::Cell;
+use std::ops::Deref;
+use std::sync::OnceLock;
+
+pub(crate) struct Lazy<T, F = fn() -> T> {
+    cell: OnceLock<T>,
+    init: Cell<Option<F>>,
+}
+
+unsafe impl<T, F> Sync for Lazy<T, F> {}
```

Comment:
If it were an `Fn() -> T + Sync` this line of unsafe would not be needed.

Reply:
```
error[E0277]: `std::cell::Cell<Option<fn() -> HashSet<KeyCode>>>` cannot be shared between threads safely
  --> src\platform_impl\windows\keyboard_layout.rs:85:25
   |
85 | static NUMPAD_KEYCODES: Lazy<HashSet<KeyCode>> = Lazy::new(|| {
   |                         ^^^^^^^^^^^^^^^^^^^^^^ `std::cell::Cell<Option<fn() -> HashSet<KeyCode>>>` cannot be shared between threads safely
   |
   = help: within `Lazy<HashSet<KeyCode>>`, the trait `Sync` is not implemented for `std::cell::Cell<Option<fn() -> HashSet<KeyCode>>>`
   = note: if you want to do aliasing and mutation between multiple threads, use `std::sync::RwLock`
note: required because it appears within the type `Lazy<HashSet<KeyCode>>`
  --> src\utils.rs:8:19
   |
8  | pub(crate) struct Lazy<T, F: Fn() -> T + Sync = fn() -> T> {
   |                   ^^^^
   = note: shared static variables must have a type that implements `Sync`
```
The issue primarily stems from the Cell on the inside, but said cell is only accessed if and only if the OnceLock has not been initialized, so the unsafe impl is indeed still sound so long as T and F is Send.

Reply:
One way around this is just to use `Mutex` in place of the `Cell`, and eat the initialization cost once.

Reply:
Nevermind, I misunderstood your comment. Switched to using a function pointer instead of a generic over potential closures.